src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -128,28 +132,34 @@ public boolean hasRentalInformation(Client client, RentalInformation rentalInfor
         return addressBook.hasRentalInformation(client, rentalInformation);
     }
 
-    //=========== Filtered Client List Accessors =============================================================
+    //=========== Filtered and Sorted Client List Accessors ==========================================================
 
     /**
      * Returns an unmodifiable view of the list of {@code Client} backed by the internal list of
      * {@code versionedAddressBook}
      */
     @Override
     public ObservableList<Client> getFilteredPersonList() {
-        return filteredClients;
+        return sortedFilteredClients;
     }
 
     @Override
     public void updateFilteredPersonList(Predicate<Client> predicate) {
         requireNonNull(predicate);
         filteredClients.setPredicate(predicate);
 
-        if (filteredClients.isEmpty()) {
+        if (sortedFilteredClients.isEmpty()) {
             updateVisibleRentalInformationList(List.of());
             setLastViewedClient(null);
         }
     }
 
+    @Override
+    public void updateFilteredPersonList(Comparator<Client> comparator) {
+        requireNonNull(comparator);
+        sortedFilteredClients.setComparator(comparator);
+    }
+
```

Comment:
I roughly know what you are trying to achieve, and it will be good to have a clearer method names for these two methods, differentiate by type of parameter may not be ideal.